react/components/EditorContainer/Sidebar/ConfigurationList/ContentEditor/ConditionControls/Scheduler.tsx

```diff
@@ -0,0 +1,239 @@
+
+    this.minDate = isLastTime(initialTimeLimit)
+      ? getNextDay(this.currDate)
+      : this.currDate
+
+    const fromMinTime = getNextTime(initialTimeLimit)
+    const toMinTime = getNextTime(fromMinTime)
+
+    const from = props.initialValues && props.initialValues.from
+    const to = props.initialValues && props.initialValues.to
+
+    this.initialState = {
+      from,
+      fromMinTime,
+      shouldDisplayFrom: !!from,
+      shouldDisplayTo: !!to,
+      to,
+      toMinTime,
+    }
+
+    this.state = this.initialState
+  }
+
+  public render() {
+    const { intl, runtime: iframeRuntime } = this.props
+
+    return (
+      <>
+        <ConditionTitle labelId="pages.editor.components.condition.date.title" />
+
+        <>
+          <Toggle
+            checked={this.state.shouldDisplayFrom}
+            label={intl.formatMessage({
+              id: 'pages.editor.components.condition.date.toggle.start',
+            })}
+            onChange={this.toggleFromVisibility}
+          />
+
+          {this.state.shouldDisplayFrom && (
+            <div className="flex mt4">
+              <DatePicker
+                align="right"
+                dateRangeStart={this.state.from}
+                dateRangeEnd={this.state.to}
+                direction="up"
+                isRangeStart
+                locale={iframeRuntime.culture.locale}
+                maxTime={this.defaultMaxTime}
+                minDate={this.minDate}
+                minTime={this.state.fromMinTime}
+                onChange={this.handleFromChange}
+                useTime
+                value={this.state.from}
+              />
+            </div>
+          )}
+        </>
+
+        <div className="mt6" />
+
+        <>
+          <Toggle
+            checked={this.state.shouldDisplayTo}
+            label={intl.formatMessage({
+              id: 'pages.editor.components.condition.date.toggle.end',
+            })}
+            onChange={this.toggleToVisibility}
+          />
+
+          {this.state.shouldDisplayTo && (
+            <div className="flex mt4">
+              <DatePicker
+                align="right"
+                dateRangeStart={this.state.from}
+                dateRangeEnd={this.state.to}
+                direction="up"
+                isRangeEnd
+                locale={iframeRuntime.culture.locale}
+                maxTime={this.defaultMaxTime}
+                minDate={this.minDate}
+                minTime={this.state.toMinTime}
+                onChange={this.handleToChange}
+                useTime
+                value={this.state.to}
+              />
+            </div>
+          )}
+        </>
+      </>
+    )
+  }
+
+  private handleFromChange = (from: Date) => {
+    if (from < this.initialState.fromMinTime) {
+      from.setTime(this.initialState.fromMinTime.getTime())
+    }
+
+    const to =
+      this.state.to && from > this.state.to ? getNextTime(from) : this.state.to
+
+    this.setState(
```

Comment:
This `this.setState(...). () => { this.props.updateCondition({from,to}) }` function repeats itself in most of the handlers... I think moving this to a separate function would help future changes to the callback if needed. Also DRY

Reply:
You're right! I'd actually done that, but for some reason I changed it back.